torch/functional.py

```diff
@@ -1232,6 +1233,10 @@ def norm(input, p="fro", dim=None, keepdim=False, out=None, dtype=None):  # noqa
         >>> torch.norm(d[0, :, :]), torch.norm(d[1, :, :])
         (tensor(3.7417), tensor(11.2250))
     """
+    warnings.warn((
```

Comment:
Needs to be TORCH_WARN or TORCH_WARN_ONCE in C++ for C++ API equivalency. 

TORCH_WARN can be very verbose so we typically use TORCH_WARN_ONCE.

Reply:
Well there is no C++ equivalent for `torch.norm` (which is just an alias of `torch.functional.norm`), because it's implemented here in Python. Yes, there is the `at::norm` function, but that just covers the vector norm case. `torch.linalg_norm` depends on `at::norm`, so we shouldn't put any warnings in there.

Reply:
Interesting. I agree with your thinking.